app/javascript/Patient/New/PersonalDetails/PersonalDetails.res

```diff
@@ -1,3 +1,32 @@
+type data = {
+  lsg: array<array<string>>,
+  lsg_selected: string,
+  volunteers: array<array<string>>,
+  volunteers_selected: array<array<string>>,
+  ashas: array<array<string>>,
+  asha_selected: string,
+  facility: array<array<string>>,
+  facility_selected: string,
+  reported_by: array<array<string>>,
+  reported_by_selected: string,
+}
+
+@scope("JSON") @val
+external parseJson: string => data = "parse"
+
+let getData = () => {
+  let elem =
+    Domutils.doc->Domutils.getElementById("data")->Belt.Option.getWithDefault(Js.Obj.empty())
+
+  elem["innerText"]->Domutils.replaceAll("&quot;", "\"")->parseJson
+}
```

Comment:
Use `Bs-webapi` and write explicit decode. 

This way we can figure out the exact props that are missing when an error happens.